contrib/mesos/ci/test-conformance.sh

```diff
@@ -31,4 +31,14 @@ TEST_ARGS="$@"
 
 KUBE_ROOT=$(cd "$(dirname "${BASH_SOURCE}")/../../.." && pwd)
 
-"${KUBE_ROOT}/contrib/mesos/ci/run-with-cluster.sh" KUBECONFIG=~/.kube/config hack/conformance-test.sh ${TEST_ARGS}
+TEST_CMD=""
+if [ -n "${CONFORMANCE_BRANCH}" ]; then
+	CI_SHA1=$(git log -n 1 --format="%h" HEAD contrib/mesos/ci)
+
+	TEST_CMD+="git fetch https://github.com/kubernetes/kubernetes ${CONFORMANCE_BRANCH} && "
+	TEST_CMD+="git checkout FETCH_HEAD -f && "
+	TEST_CMD+="git checkout ${CI_SHA1} contrib/mesos/ci -f && "
```

Comment:
Hmm, this leaves your working tree with modified dangling files in the index. Maybe we should clean up afterwards and reset this forced commit. I have mixed feelings about this :-/